app/controllers/requested_accounts_controller.rb

```diff
@@ -91,13 +89,14 @@ def all_requested_accounts
   def create_account(requested_account)
     creation_attempt = CreateRequestedAccount.new(requested_account, current_user)
     result = creation_attempt.result
-    return result unless result[:success]
-    # If it was successful, enroll the user in the course
-    user = creation_attempt.user
-    JoinCourse.new(course: requested_account.course,
-                   user: user,
-                   role: CoursesUsers::Roles::STUDENT_ROLE)
-    result
+    result[:result_description] = creation_attempt.result_description(result)
```

Comment:
This is still some unnecessary work... you could set the `:result_description` from within `CreateRequestedAccount` right after the response is received from the server, so that calling `result = creation_attempt.result` will mean it already includes the description.

Reply:
Done!